httptrace/client.go

```diff
@@ -122,21 +123,21 @@ type Transport struct {
 	Transport http.RoundTripper
 
 	SetName bool
+
+	// requests keeps clone request
+	requests map[*http.Request]*http.Request
+	reqMu    sync.Mutex
```

Comment:
I can't comment on the entire PR, but I have a minor style suggestion for this code segment.

It's common to put the `sync.Mutex` on top of the variables it protects, like a "hat". For example, see [here](https://gotools.org/net/http#transport-go-L55) how `Transport` defines its `sync.Mutex` fields. Sticking with that pattern helps readability.